### PR TITLE
Add support for OBJ and BINSTRING

### PR DESCRIPTION
**Pull request by suhacker1**

### CODE REVIEW

File: fickling/fickle.py [hunk 1]
```diff
@@ -1123,6 +1123,26 @@ def run(self, interpreter: Interpreter):
...
+                break
+            args.append(arg)
+        else:
+            raise ValueError("Exhausted the stack while searching for a MarkObject!")
+        kls = args.pop()
+        # TODO Verify paths for correctness
+        if args or hasattr(kls, "__getinitargs__") or not isinstance(kls, type):
+            interpreter.stack.append(ast.Call(kls, [list(args)], []))
+        else:
+            interpreter.stack.append(ast.Call(kls, kls, []))
```
Comment (Boyan-MILANOV) [inline fickling/fickle.py:1143]: The arguments will be in the wrong order here. They are appended to the list from last to first, so we should invert the list order to recover the actual order of arguments.

```suggestion
    def run(self, interpreter: Interpreter):
        args = []
        while interpreter.stack:
            arg = interpreter.stack.pop()
            if isinstance(arg, MarkObject):
                break
            args.insert(0, arg)
        else:
            raise ValueError("Exhausted the stack while searching for a MarkObject!")
        kls = args.pop(0)
        # TODO Verify paths for correctness
        if args or hasattr(kls, "__getinitargs__") or not isinstance(kls, type):
            interpreter.stack.append(ast.Call(kls, args, []))
        else:
            interpreter.stack.append(ast.Call(kls, kls, []))
```

**Review by Boyan-MILANOV — CHANGES_REQUESTED** [review]

File: fickling/fickle.py [hunk 2]
```diff
@@ -1469,6 +1489,22 @@ def validate(cls, obj):
         return obj
 
 
+class BinString(DynamicLength, ConstantOpcode):
+    name = "BINSTRING"
+    priority = ShortBinBytes.priority + 1
+    length_bytes = 4
+    signed = True
```
[the remark this answers is not in the document]
Comment (suhacker1) [reply fickling/fickle.py:1496]: This is the source code for that opcode. I may have interpreted this incorrectly. 
```
    def load_binstring(self):
        # Deprecated BINSTRING uses signed 32-bit length
        len, = unpack('<i', self.read(4))
        if len < 0:
            raise UnpicklingError("BINSTRING pickle has negative byte count")
        data = self.read(len)
        self.append(self._decode_string(data))
    dispatch[BINSTRING[0]] = load_binstring
```

File: test/test_crashes.py [hunk 3]
```diff
@@ -62,3 +62,7 @@ def test_pop_mark(self):
         """Tests the correctness of the POP_MARK opcode by using the bytecode from https://github.com/mindspore-ai/mindspore/issues/183
         This can be simplified to allow for the correctness of additional opcodes to be tested"""
         pass
+
+    @unparse_test(io.BytesIO(b'(cos\nsystem\nS"whoami"\no.'))
+    def test_obj(self):
+        pass
```
[the remark this answers is not in the document]
Comment (suhacker1) [reply test/test_crashes.py:68]: This is just the input that raised the error about OBJ not being implemented. I can remove this if it's superfluous or incorrect.

File: fickling/fickle.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Boyan-MILANOV) [reply fickling/fickle.py:1496]: Oh nice catch, you were indeed correct. Why the hell would they use a signed int for the length of a string 😂

File: test/test_crashes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (suhacker1) [reply test/test_crashes.py]: Comment added!

**Review by Boyan-MILANOV — APPROVED** [review]

**Pull request merged**